systemtests/src/test/java/io/enmasse/systemtest/TestBase.java

```diff
@@ -48,73 +48,57 @@
-            : new AddressSpace("default", environment.namespace());
 
-    protected static final OpenShift openShift = new OpenShift(environment, environment.namespace(), defaultAddressSpace.getNamespace());
+
+    protected static final OpenShift openShift = new OpenShift(environment, environment.namespace());
     private static final GlobalLogCollector logCollector = new GlobalLogCollector(openShift,
             new File(environment.testLogDir()));
+    protected static final AddressApiClient addressApiClient = new AddressApiClient(openShift);
+
     private KeycloakClient keycloakApiClient;
-    protected AddressApiClient addressApiClient;
     protected String username;
     protected String password;
     protected AmqpClientFactory amqpClientFactory;
     protected MqttClientFactory mqttClientFactory;
+    protected List<AddressSpace> addressSpaceList = new ArrayList<>();
 
-    protected boolean createDefaultAddressSpace() {
-        return true;
-    }
+    protected AddressSpace getSharedAddressSpace() { return null; }
 
     @Before
     public void setup() throws Exception {
-        addressApiClient = new AddressApiClient(openShift);
-        if (createDefaultAddressSpace()) {
-            if (environment.isMultitenant()) {
-                Logging.log.info("Test is running in multitenant mode");
-                createAddressSpace(defaultAddressSpace, environment.defaultAuthService());
-                // TODO: Wait another minute so that all services are connected
-                Logging.log.info("Waiting for 2 minutes before starting tests");
-            }
-
-            if ("standard".equals(environment.defaultAuthService())) {
-                this.username = "systemtest";
-                this.password = "systemtest";
-                getKeycloakClient().createUser(defaultAddressSpace.getName(), username, password, 1, TimeUnit.MINUTES);
-            }
-            amqpClientFactory = new AmqpClientFactory(openShift, environment, defaultAddressSpace, username, password);
-            mqttClientFactory = new MqttClientFactory(openShift, environment, defaultAddressSpace, username, password);
-        }
+        addressSpaceList = new ArrayList<>();
+        amqpClientFactory = new AmqpClientFactory(openShift, environment, getSharedAddressSpace(), username, password);
+        mqttClientFactory = new MqttClientFactory(openShift, environment, getSharedAddressSpace(), username, password);
     }
 
     @After
     public void teardown() throws Exception {
-        if (mqttClientFactory != null) {
-            mqttClientFactory.close();
-        }
-        if (amqpClientFactory != null) {
-            amqpClientFactory.close();
-        }
-        if (addressApiClient != null) {
-            if (createDefaultAddressSpace()) {
-                setAddresses();
-            }
-            addressApiClient.close();
```

Comment:
addressApiClient is never closed, right?

Reply:
No, but it is static now so it is only instantiated once throughout the entire testrun.